OpenRA.Mods.Common/Traits/Attack/AttackSuicides.cs

```diff
@@ -14,30 +14,31 @@
 using OpenRA.Mods.Common;
 using OpenRA.Mods.Common.Activities;
 using OpenRA.Mods.Common.Orders;
-using OpenRA.Mods.Common.Traits;
 using OpenRA.Traits;
 
-namespace OpenRA.Mods.RA.Traits
+namespace OpenRA.Mods.Common.Traits
 {
-	class DemoTruckInfo : ITraitInfo, Requires<ExplodesInfo>
+	[Desc("Does a suicide attack where it moves next to the target when used in combination with `Explodes`.")]
+	class AttackSuicidesInfo : ITraitInfo
 	{
 		[VoiceReference] public readonly string Voice = "Action";
 
-		public object Create(ActorInitializer init) { return new DemoTruck(init.Self, this); }
+		public object Create(ActorInitializer init) { return new AttackSuicides(init.Self, this); }
 	}
 
-	class DemoTruck : IIssueOrder, IResolveOrder, IOrderVoice
+	class AttackSuicides : IIssueOrder, INotifyCreated, IResolveOrder, IOrderVoice
 	{
-		readonly DemoTruckInfo info;
+		readonly AttackSuicidesInfo info;
+		IMove move;
 
-		public DemoTruck(Actor self, DemoTruckInfo info)
+		public AttackSuicides(Actor self, AttackSuicidesInfo info)
 		{
 			this.info = info;
 		}
 
-		static void Explode(Actor self)
+		public void Created(Actor self)
 		{
-			self.World.AddFrameEndTask(w => self.InflictDamage(self, int.MaxValue, null));
+			move = self.Trait<IMove>();
```

Comment:
You may as well move this into the constructor since you need the `Requires` anyway.